core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -66,6 +68,8 @@ class LogSegment(val log: FileRecords,
   /* The maximum timestamp we see so far */
   @volatile private var maxTimestampSoFar = timeIndex.lastEntry.timestamp
   @volatile private var offsetOfMaxTimestamp = timeIndex.lastEntry.offset
+  /* The offset that would be used for the next message to be appended to this segment */
+  @volatile private var _nextOffset = INVALID_NEXT_OFFSET
```

Comment:
Yes, that works. It's just that if we initialize to baseOffset (which is what nextOffsetFromLog() will return on an empty segment), _nextOffset is always initialized properly and we probably can get rid of the if test in nextOffset().